File: src/Swift/task_1/pattern.swift
import Foundation

func printAlphabetPyramid(_ N: Int) {
    // Итерация от N до 1 для печати строк сверху вниз
    for row in (1...N).reversed() {
        // Итерация от N до row для печати символов в строке справа налево
        for letter in (row...N).reversed() {
            // Печать символа, начиная с соответствующего номеру строки
            let scalar = UnicodeScalar(64 + letter)! // ASCII значение для 'A' - 65, но тут 64 + letter, т.к. индексация начинается с 1
            print("\(scalar) ", terminator: "")
        }
        // Переход на новую строку
        print("")
    }
}

// Чтение значения N
if let N = Int(readLine() ?? "0") {
    printAlphabetPyramid(N)
}

File: src/Ruby/task_1/pattern.rb
def print_alphabet_pyramid(n)
    # Итерация от n до 1 для печати строк сверху вниз
    n.downto(1) do |row|
      # Итерация от n до row для печати символов в строке справа налево
      n.downto(row) do |letter|
        # Печать символа, начиная с соответствующего номеру строки
        print (65 + letter - 1).chr + ' '
      end
      # Переход на новую строку
      puts
    end
  end
  
  n = gets.to_i
  print_alphabet_pyramid(n)